pennylane/interfaces/autograd.py

```diff
@@ -42,8 +41,23 @@ def interface(self):
         # mark the evaluate method as an Autograd primitive
         evaluate = autograd.extend.primitive(qnode.__class__.evaluate)
 
+        def set_trainable(self, args):
+            """Given input arguments to the AutogradQNode, determine which arguments
+            are trainable and which aren't.
+
+            This method calls the underlying :meth:`set_trainable_args` method of the QNode.
+            """
+            non_trainable_args = set()
+
+            for idx, arg in enumerate(args):
+                if not getattr(arg, "requires_grad", True):
+                    non_trainable_args.add(idx)
+
+            self.set_trainable_args(set(range(len(args))) - non_trainable_args)
```

Comment:
why not just have a `trainable_args` set instead?

Reply:
Good question 🤔

Reply:
Have changed this to `trainable_args`